internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -366,6 +389,12 @@ func (m *BlobMetadataManager) createFileFinalize(filePath string, fileMetadata [
 	if common.IsDebugBuild() {
 		prop, err := m.storageCallback.GetPropertiesFromStorage(
 			internal.GetAttrOptions{Name: path})
+
+		injectErr := common.InjectError(common.PROB_MODERATE)
+		if injectErr != nil {
+			err = injectErr
+		}
+
```

Comment:
don't inject error here as it's only for debug builds.
also you have an assert fail right after this.

**As I said, it's very important you make sure each of the error injections you add are verified.**